allen-cell-animated/website-3d-cell-viewer
# feature: Loading from the landing page, viewer page routing

*Estimated review size: medium, 30-40 minutes*

Closes #188, the landing page implementation!

This change finishes implementing loading from the landing page and also sets up routing for the app.

These changes seem large, but the majority of the changes are just file moves and renames! I'll note what files you can compare between, and a summary of any changes on each file.

## Changes:
- Adds `react-router-dom` and routes for the landing page (at root URL, `/`) and viewer (`/viewer`).
- The landing page can now redirect to the viewer when a Load button is clicked, and passes the app arguments as state through the navigation API.
- Renamed the `landing-page` directory to `website`, since it now includes several website-related components.
- Adds a new `AppWrapper` component, which wraps `ImageViewerApp`.
  - Handles retrieving app arguments from URL or from routing.
- Moves the URL parsing logic to `url_utils.tsx`.

### Videos (🔊)

Browser Router:

https://github.com/allen-cell-animated/website-3d-cell-viewer/assets/30200665/176cc000-384f-4e75-a5cf-8cc471499f0a

Hash Router:

https://github.com/allen-cell-animated/website-3d-cell-viewer/assets/30200665/51920aa7-e427-4793-b973-4cc85ff7fd72

Loading empty viewer:
![image](https://github.com/allen-cell-animated/website-3d-cell-viewer/assets/30200665/2c880111-2867-4196-a635-f9e64dfc2de4)

### Validation:
1. Run the app locally. The default URL should open on the landing page.
2. Click on the "Load" button, and it should open the volume in the `/viewer` page.
3. Refreshing the page should still show the same volume.
4. Go to http://localhost:9020/viewer?url=https://animatedcell-test-data.s3.us-west-2.amazonaws.com/20200323_F01_001/P13-C4.zarr . This should also open correctly in the browser.
5. Go to http://localhost:9020?url=https://animatedcell-test-data.s3.us-west-2.amazonaws.com/20200323_F01_001/P13-C4.zarr . This will redirect to the `/viewer` page.
